kong/plugins/prometheus/exporter.lua

```diff
@@ -168,6 +174,50 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- upstream targets accessible?
+  local balancer = require "kong.runloop.balancer"
+  local upstreams_dict = balancer.get_all_upstreams()
+  for upstream_name, upstream_id in pairs(upstreams_dict) do
+    -- same logic from kong.db.dao.targets
+    local health_info
+    health_info, err = balancer.get_upstream_health(upstream_id)
+    if err then
+      ngx.log(ngx.ERR, "failed getting upstream health: ", err)
```

Comment:
let's use pdk here `kong.log.err("failed getting upstream health: ", err)` for better prefix in log

Reply:
ok, I'll change it too. But this is how kong.db.dao.target implements it.